### PR TITLE
Handle SymrefLoop

### PR DESCRIPTION
**Pull request by jelmer**
Handle SymrefLoop. Fixes: #293

### CODE REVIEW

File: klaus/repo.py [hunk 1]
```diff
@@ -136,9 +142,14 @@ def get_default_branch(self):
...
             except KeyError:
                 pass
-        try:
-            return self.get_branch_names()[0]
-        except IndexError:
+            except SymrefLoop:
+                pass
+        for name in self.get_branch_names():
+            try:
+                self.get_commit(name)
```
Comment (jonashaag) [inline klaus/repo.py:149]: Missing return

**Comment by jonashaag** [comment]
Why aren’t actions running? :(

**Comment by jonashaag** [comment]
Can you please rebase latest master

**Comment by jonashaag** [comment]
WDYT about having the tuple `(KeyError, SymrefLoop)` defined as a constant somewhere and using that instead of repeating the list of exceptions everywhere? Could also have a short comment there what’s SymrefLoop.

**Comment by jelmer** [comment]
> WDYT about having the tuple `(KeyError, SymrefLoop)` defined as a constant somewhere and using that instead of repeating the list of exceptions everywhere? Could also have a short comment there what’s SymrefLoop.

Happy to define a shorthand for those two places where we catch both (InaccessibleRef, perhaps?). That said, I'm wary of using that everywhere - e.g. in views.py where a message to the user saying a ref doesn't exist is quite different from saying the ref is broken.

**Comment by jelmer** [comment]
Updated.

**Comment by jonashaag** [comment]
Perfect!

**Pull request merged**